lxd/cluster/gateway.go

```diff
@@ -234,24 +238,30 @@ func (g *Gateway) HandlerFuncs(heartbeatHandler HeartbeatHandler, trustedCerts f
 
 		// From here on we require that this node is part of the raft
 		// cluster.
+		g.lock.RLock()
 		if g.server == nil || g.memoryDial != nil {
+			g.lock.RUnlock()
 			http.NotFound(w, r)
 			return
 		}
+		g.lock.RUnlock()
 
 		// NOTE: this is kept for backward compatibility when upgrading
 		// a cluster with version <= 4.2.
 		//
 		// Once all nodes are on >= 4.3 this code is effectively
 		// unused.
 		if r.Method == "HEAD" {
+			g.lock.RLock()
```

Comment:
I think for this type of request we can use: 

```
g.lock.RLock()
defer g.lock.RUnlock()
```

As we intentionally want to hold the read lock for the whole HEAD request until it returns.

Reply:
yeah, that's nicer, all those `RUnlock()` were hurting already my eyes.